### PR TITLE
Automated testing with protractor and travis

### LINKED ISSUE #68: Automated testing
In the [old zonemaster-gui] we had Travis run our automated test suite (see .travis.yml). When we switched to the new zonemaster-gui we lost this feature. It should be reimplemented.

See also #26, #37.

N.B. Don't forget to update the "Update CI configuration" section of the [ReleaseProcess].

[old zonemaster-gui]: https://github.com/dotse/zonemaster-gui/
[ReleaseProcess]: https://github.com/zonemaster/zonemaster/blob/master/docs/internal-documentation/maintenance/ReleaseProcess.md

### PR DESCRIPTION
**Pull request by pamasse**
Based on the old_zonemaster repo, i just translate some perl tests in protractor.
Protractor is a JavaScript framework on top of Selenium which run a headless chrome/firefox and navigate (click, change page, use form ...) on a webapp to test it.

I also add a .travis.yml to run tests before each PR.
The last two tests (predelegated and simple test) are end-2-end test and depends on zonemaster.net.

Run a test take around 5min.

fix #68

### CODE REVIEW

File: e2e/GR08.e2e-spec.ts [hunk 1]
```diff
@@ -2,23 +2,18 @@ import { by, browser, element } from 'protractor';
 
 import { Utils } from './pages/app.utils';
 
-describe('Zonemaster test GR08 - [The advanced view should support the possibility of choosing a profile from multiple profiles]', () => {
+describe('Zonemaster test GR08 - [The advanced view should look the same in latest version of different browsers]', () => {
```
[the remark this answers is not in the document]
Comment (pamasse) [reply e2e/GR08.e2e-spec.ts]: Thx for the link ! I do a first version of tests without any specifications. I will do a second PR (or just improve this PR) with an update of the functional test requirements file and with new tests align with it this time.  

The main issue here is the need of a functional zonemaster backend. I will certainly provide a mock for testing.

[the remark this answers is not in the document]
Comment (mattias-p) [reply e2e/GR08.e2e-spec.ts]: Sounds good! Do you want to create issues about getting the requirements file and the test implementations in sync? Or if you like, I can create it/them.

I like the idea of mocking the RPCAPI as well. We should specify the outputs of the mocked methods too, to make it clear what behavior the mock is supposed to emulate.

In any case I'd be OK with merging this as and make the improvements later. As long as we don't forget them.

File: e2e/GR10.e2e-spec.ts [hunk 2]
```diff
@@ -2,16 +2,26 @@ import { by, browser, element } from 'protractor';
 
 import { Utils } from './pages/app.utils';
 
-describe('Zonemaster test GR10 - [The preDelegatedDomainCheck view should have a shortcut to undelegated view and FAQ]', () => {
+describe('Zonemaster test GR10 - [The undelegated view should have a text describing what undelegated means?]', () => {
```
[the remark this answers is not in the document]
Comment (matsduf) [reply e2e/GR10.e2e-spec.ts]: The document [undelegated-test] defines an undelegated test. Does that work?

[undelegated-test]: https://github.com/zonemaster/zonemaster/blob/master/docs/specifications/test-types/undelegated-test.md

File: e2e/GR08.e2e-spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (pamasse) [reply e2e/GR08.e2e-spec.ts]: Issue #83

**Pull request merged**